lang/typescript.talon

```diff
@@ -153,3 +220,39 @@ state reduce:
   key(left)
 
 state spread: "..."
+
+^state export class <user.text>$:
+  insert("export class  {}")
+  edit.left()
+  key(enter)
+  edit.up()
+  edit.line_end()
+  key(left left)
+  user.code_type_formatter(user.text)
+
+^state extends <user.text>$:
+  insert(" extends ")
+  user.code_type_formatter(user.text)
+
+^state implements <user.text>$:
+  insert(" implements I")
+  user.code_interface_formatter(user.text)
+
+^state param <user.text>$:
+  user.code_private_variable_formatter(user.text)
+  insert(": ")
+
+^state next param <user.text>$:
+  insert(", ")
+  user.code_private_variable_formatter(user.text)
+  insert(": ")
+
+state return default:
+  edit.right()
+  insert(": ")
+  insert("void")
+
+^state return type <user.text>$:
+  edit.right()
+  insert(": ")
+  user.code_type_formatter(user.text)
```

Comment:
Are types in typescript uniformly public camel case? I'm completely unfamiliar with it, just a dumb question.

This one is beyond this PR, just picking your brain. What do you think about adding a capture for some types in code.py, similar to what manwar's done for C? 
https://github.com/ma-anwar/knausj_talon/blob/bb893df3ede9f77944903c6c98139ecead1a5d70/code/c.py#L17

Reply:
Both js and ts *should* use PascalCase for modules / types and camelCase for members. The exceptions are aliases for primitives (as is common in many languages) like `string`, `boolean`, `number`, etc. So yes, some combination of these may work really well together.

I was going to solve that with two commands, but that may be ugly. Something like `state return type` and `state return primitive`

Reply:
Thanks. A combo seems like a reasonable approach here, I'll look into the captures sometime this week hopefully.

FYI, I did some prelim work on snippet support, which you can likely use easily with your rewrite fwiw
https://github.com/knausj85/knausj_talon/commit/1e4eea598e795c77da960da4e0880baddd4fa2f7